libraries/freertos_plus/standard/freertos_plus_tcp/source/portable/NetworkInterface/Zynq/x_emacpsif_physpeed.c

```diff
@@ -338,32 +334,32 @@ unsigned get_IEEE_phy_speed(XEmacPs *xemacpsp)
 			break;
 	}
 #endif
-	xil_printf("Waiting for PHY to complete autonegotiation.\r\n");
+	FreeRTOS_printf( ( "Waiting for PHY to complete autonegotiation.\n" ) );
 
 	XEmacPs_PhyRead(xemacpsp, phy_addr, IEEE_STATUS_REG_OFFSET, &status);
 	while ( !(status & IEEE_STAT_AUTONEGOTIATE_COMPLETE) ) {
-		sleep(1);
+		vTaskDelay( 2 );
```

Comment:
Should this delay value be a configurable #define?

Reply:
+1.  Also, how does this code handle a plug/unplug during operation?

Reply:
I added a define `MINIMUM_SLEEP_TIME` with a value of 2.
As you know, a `vTaskDelay(1)` could return witin a few µS when called just before a clock-tick.
`